Purchases/SwiftSources/LocalReceiptParsing/BasicTypes/AppleReceipt.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  AppleReceipt.swift
+//  Purchases
+//
+//  Created by Andrés Boedo on 7/22/20.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+struct ReceiptAttribute {
+    let type: ReceiptAttributeType
+    let version: UInt
+    let value: String
+}
+
+enum ReceiptAttributeType: UInt {
+    case bundleId = 2,
+         applicationVersion = 3,
+         opaqueValue = 4,
+         sha1Hash = 5,
+         creationDate = 12,
+         inApp = 17,
+         originalApplicationVersion = 19,
+         expirationDate = 21
+}
+
+struct AppleReceipt {
+    let bundleId: String
+    let applicationVersion: String
+    let originalApplicationVersion: String
+    let opaqueValue: Data
+    let sha1Hash: Data
+    let creationDate: Date
+    let expirationDate: Date?
+    let inAppPurchases: [InAppPurchase]
+
+    func purchasedIntroOfferProductIdentifiers() -> Set<String> {
```

Comment:
I think I already mentioned in the other PR but this is actually checking if it's in into offer or in trial period but the name only indicates it checks if it's in into offer

Reply:
I had done it this way intentionally at first because Apple seems to be trying to unify the two concepts together into intro offers - free trials are configured as a type of intro offer with price = 0. 
However, in the receipt, there are two fields, with `isInFreeTrialPeriod` being undocumented, but it's the only one with the correct value for free trials (sigh). 
so yeah, It might make sense to rename.